File: src/main/java/shopping/domain/cart/Product.java
package shopping.domain.cart;

import javax.persistence.*;

@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false, length = 255)
    private String name;

    @Column(name = "image_url", nullable = false, length = 255)
    private String imageUrl;

    @Embedded
    @AttributeOverride(name = "price", column = @Column(name = "price", nullable = false))
    @AttributeOverride(name = "type", column = @Column(name = "type", nullable = false))
    private Money price;

    public Product(String name, String imageUrl, Double price) {
        this.name = name;
        this.imageUrl = imageUrl;
        this.price = new Money(price);
    }

    public Product(Long id, String name, String imageUrl, Double price) {
        this(name, imageUrl, price);
        this.id = id;
    }

    protected Product() {

    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public Money getPrice() {
        return price;
    }
}

File: src/main/kotlin/shopping/core/product/Product.kt
package shopping.core.product

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EntityListeners
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.time.LocalDateTime

@Table(name = "products")
@Entity
@EntityListeners(AuditingEntityListener::class)
class Product(
    @Column(nullable = false, length = 15)
    val name: String = "",

    @Column(nullable = false)
    val price: Long = 0,

    @Column(nullable = false, length = 500)
    val imageUrl: String = "",

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,
) {
    @CreatedDate
    @Column(nullable = false, updatable = false)
    var createdAt: LocalDateTime = LocalDateTime.now()
        protected set

    @LastModifiedDate
    @Column(nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now()
        protected set
}
